### PR TITLE
Add minio patches for tekton-results api server

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
#What does this pull request do

Add minio patches for tekton-results api server:
  - Add workaround to fix creation minio tenant using old(4.5.4) and newer(4.5.8) minio operator.
  - Add tekton-results api server deployment patches to make connection with minio tenant.
  - Handle minio uninstallation with help of reset.sh script.
  - Fix reset.sh script uninstallation pipeline-service, but only with reset-hard mode enabled.

### CODE REVIEW

**Comment by Roming22** [comment]
@AndrienkoAleksandr Adding `rsync-3.2.3` to the list of packages installed in `operator/images/cluster-setup/Dockerfile` should solve the test failure.

**Comment by Roming22** [comment]
1. I gave you the wrong Dockerfile. It should have been `ci/images/ci-runner`.
2. The change has to be done in another PR, so the image used during the ci already has the software installed. I've submitted #464. Once that PR is merged, I'll rerun the tests.
3. You'll need to squash the commits . You can wait for all the tests to be successful if you wish.

**Comment by AndrienkoAleksandr** [comment]
@Roming22 we need one more envsubst dependency.

**Comment by Roming22** [comment]
@AndrienkoAleksandr Latest run is failing because of minio. I'll let you correct the issue, as the error is now related to the PR.

**Comment by AndrienkoAleksandr** [comment]
@Roming22 ci is green https://github.com/openshift-pipelines/pipeline-service/pull/461/checks?check_run_id=11184105678 for commit https://github.com/openshift-pipelines/pipeline-service/pull/461/commits/315941740d73b86da953b67d880c9e041070c110

**Comment by Roming22** [comment]
Please squash all commits into a single one. Make sure the commit message is helpful.

**Comment by AndrienkoAleksandr** [comment]
I provided one more extra commit [This commit must be delete before merge.](https://github.com/openshift-pipelines/pipeline-service/pull/461/commits/e707c3e167cc60ac4b2dcc544106cbb1ddfbb117) because I don't know if ci is handling repository name.

**Comment by Roming22** [comment]
@AndrienkoAleksandr Please squash all the commits (including mine) into a single one. Since we've seen that the CI is OK, remove the commit that references your fork.
Next run will be red, but I'll merge the PR.

**Comment by AndrienkoAleksandr** [comment]
> @Roming22: Please squash all the commits (including mine) into a single one.

Done.

File: developer/config.yaml [hunk 1]
```diff
@@ -9,10 +9,12 @@ cluster_type: openshift
 git_url: https://github.com/openshift-pipelines/pipeline-service.git
 
 # git_ref refers to the git repo's ref to be considered as the source of truth for Argo CD applications.
-git_ref: main
+git_ref: dev-kustomization
```
Comment (Roming22) [inline developer/config.yaml:12]: Revert to `main`.

**Review by Roming22 — COMMENTED** [review]
@AndrienkoAleksandr Sorry, one last small update is needed before I can approve.

File: developer/config.yaml (hunk 1 above)
Comment (AndrienkoAleksandr) [reply developer/config.yaml]: Oh, nice catch thanks. I will fix it in all places.

**Comment by AndrienkoAleksandr** [comment]
> @AndrienkoAleksandr Sorry, one last small update is needed before I can approve.

done.

**Pull request merged**